gulpfile.js

```diff
@@ -0,0 +1,870 @@
+
+var buildFiles = ['Specs/**/*.js',
+                  '!Specs/SpecList.js',
+                  'Source/Shaders/**/*.glsl'];
+
+var jsHintFiles = ['Source/**/*.js',
+                   '!Source/Shaders/**',
+                   '!Source/ThirdParty/**',
+                   '!Source/Workers/cesiumWorkerBootstrapper.js',
+                   'Apps/**/*.js',
+                   'Apps/Sandcastle/gallery/*.html',
+                   '!Apps/Sandcastle/ThirdParty/**',
+                   'Specs/**/*.js',
+                   'Tools/buildTasks/**/*.js',
+                   'gulpfile.js'];
+
+var filesToClean = ['Source/Cesium.js',
+                    'Build',
+                    'Instrumented',
+                    'Source/Shaders/**.js',
+                    'Specs/SpecList.js',
+                    'Apps/Sandcastle/.jshintrc',
+                    'Apps/Sandcastle/jsHintOptions.js',
+                    'Apps/Sandcastle/gallery/gallery-index.js',
+                    'Cesium-*.zip'];
+
+var filesToSortRequires = ['Source/**/*.js',
+                           '!Source/Shaders/**',
+                           '!Source/ThirdParty/**',
+                           '!Source/Workers/cesiumWorkerBootstrapper.js',
+                           '!Source/copyrightHeader.js',
+                           '!Source/Workers/transferTypedArrayTest.js',
+                           'Apps/**/*.js',
+                           '!Apps/Sandcastle/ThirdParty/**',
+                           '!Apps/Sandcastle/jsHintOptions.js',
+                           'Specs/**/*.js',
+                           '!Specs/spec-main.js',
+                           '!Specs/SpecRunner.js',
+                           '!Specs/SpecList.js',
+                           '!Apps/Sandcastle/Sandcastle-client.js',
+                           '!Apps/Sandcastle/Sandcastle-header.js',
+                           '!Apps/Sandcastle/Sandcastle-warn.js',
+                           '!Apps/Sandcastle/gallery/gallery-index.js'];
+
+gulp.task('default', ['combine']);
+
+gulp.task('build', function(done) {
+    mkdirp.sync('Build');
+    glslToJavaScript(false, 'Build/minifyShaders.state');
+    createCesiumJs();
+    createSpecList();
+    createGalleryList();
+    createSandcastleJsHintOptions();
+    done();
+});
+
+gulp.task('build-watch', function() {
+    gulp.watch(buildFiles, ['build']);
+});
+
+gulp.task('buildApps', ['combine', 'minifyRelease'], function() {
+    return buildCesiumViewer();
+});
+
+gulp.task('clean', function(done) {
+    async.forEach(filesToClean, rimraf, done);
+});
+
+gulp.task('cloc', ['build'], function() {
+    var glsl = globby.sync(['Source/Shaders/*.glsl', 'Source/Shaders/**/*.glsl', 'Source/main.js']);
+    glsl = glsl.join(' ');
+
+    var clockPath = path.join('Tools', 'cloc-1.60', 'cloc-1.60.pl');
+    var cloc_definitions = path.join('Tools', 'cloc-1.60', 'cloc_definitions');
+
+    var cmdLine;
+    return Promise.join(
```

Comment:
The callback isn't called until the process terminates and stdout and stderr will be buffers that contain the complete output of the function, so there's no way for them to be jumbled.  However, it is possible for the Specs section to print out before the Source section, so I'll chain them together just so the output is always guaranteed the same.